### PR TITLE
Add check for rundeck_acl_policy name

### PR DESCRIPTION
**Pull request by JGodin-C2C**
##### SUMMARY
This change simply introduce a sanity check over the name of the policy. 
Using this module and using a name that is not compatible with rundeck prerogatives lead to a cryptic error 
`"Unhandled HTTP status -1, please report the bug"`

This pull request will check if the name passed to the module passes rundeck name policy. If not, it will output a more understandable error.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME

rundeck_acl_policy

###### How to reproduce ?

Simply execute this task on any rundeck of your choice. 

```
- name: "Create or update a rundeck ACL policy for users to be able to read"
  rundeck_acl_policy:
    name: "Name with forbidden characters ( here , spaces for exemple )"
    api_version: 21
    url: "https://my.rundeck"
    token: "{{ rundeck_api_token }}"
    state: present
    project: "allow_users_for_project_{{ item.name }}"
    policy:
      description: "normal users will only have read permissions for {{ item.name }}"
      context:
        application: rundeck
      for:
        project:
          - match:
              name: "{{ item.name }}"
            allow: [read]
        system:
          - match:
              name: '.*'
            allow: [read]
        storage:
          - match:
              path: 'keys/.*'
            allow: [read]
      by:
        username: "{{ item.users | list }}"

```

### CODE REVIEW

**Comment by JGodin-C2C** [comment]
Hello, 
I do not understand the errors found by shippable. 
Can anyone provide some help ? 
J\x00

**Comment by felixfontein** [comment]
I've restarted the tests, hopefully they will be more helpful now.

File: changelogs/fragments/add_argument_check_for_rundeck.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- rundeck_acl_policy - Add check for rundeck_acl_policy name parameter
```
Comment (Andersson007) [inline changelogs/fragments/add_argument_check_for_rundeck.yaml:2]:
```suggestion
- rundeck_acl_policy - add check for rundeck_acl_policy name parameter (https://github.com/ansible-collections/community.general/pull/612).
```
according to the doc

**Comment by felixfontein** [comment]
@JGodin-C2C do you have a link to the documentation stating that names should be of this form?

(I don't know rundeck at all, but I'm happy to merge this if I can see some reference that this actually works ;) )

**Comment by JGodin-C2C** [comment]
I could not find any reference of this in the documentation. 

However, i have screenshot of the "manual" ACL creation screen : 
Screenshot from 2020-07-15 10-17-16

**Comment by Andersson007** [comment]
@felixfontein ^

**Comment by felixfontein** [comment]
@JGodin-C2C then let's hope they kept that UI info up to date ;)

**Comment by felixfontein** [comment]
@JGodin-C2C thanks a lot for fixing this!
@Andersson007 thanks for reviewing!

**Pull request merged**